### ISSUE TITLE
Ability to "remove" images from AugmentedImageDatabase

### ISSUE THREAD

**Issue by achuvm** [google-ar/arcore-android-sdk#372]
https://developers.google.com/ar/reference/java/com/google/ar/core/AugmentedImageDatabase

From what we can tell there's only an `addImage*` API which means if I were to add 5 images to "find", and then decided I want to remove one, the way to do that would be to create a new `AugmentedImageDatabase` and then add the other 4 images I want to continue to augment then set the new `AugmentedImageDatabase` on the Config?

**Comment by inio** [comment]
Feature request filed.  Thanks!

**Comment by mingcen** [comment]
Hey achuvm, it is unnecessary to always rebuild the database, up to your use case. For example, if the goal is to not track a "removed" image, you might keep a list of removed image in your app, and skip the corresponding result from getUpdateTrackable<AugmetendImage>().

**Comment by achuvm** [comment]
Does it not impact performance if the underlying ARCore image tracking/detection library keeps on tracking the images I no longer care about?

If we're attempting to build a story-book like recognition app, we'd want to find maybe 5 images at a time in a sliding-window type algorithm. If the book's got 100 pages, then through your "solution" we'd end up with 100 pages, or we'd have to rebuild the database 95 times...

**Comment by inio** [comment]
My understanding is that augmented image detection works somewhat like a hash table, so the overhead as the size of the database grows is non-linear.

**Comment by inio** [comment]
Talked to the Augmented Images team a bit, and got some details:
* The runtime compute cost of augmented images detection is substantially sub-linear (nearly log(n)) on the number of images in the database.
* The runtime compute cost of `addImage()` is based on the image size and content
* The runtime compute cost of `configure()` with an updated augmented image database is roughly n·log(n) on the number of images in the database.

**Comment by inio** [comment]
Given that, I would suggest creating a database containing the images you want to detect all the time.  Serialize this database to memory.  Keep a live database that you add images to until it grows beyond a certain size, at which point you discard it, deserialize the original database, and add just the small number of images you currently care about detecting.

**Comment by markus2610** [comment]
@inio And what if I want to use two imgdb? One group of images is always detected and the second later at runtime. I want to merge both files.

**Comment by markus2610** [comment]
@achuvm How is the database rebuild? Add images and call ArConfig_setAugmentedImageDatabase, or I have to note something?

**Comment by achuvm** [comment]
@markus2610 yeah, you'll need to recreate the AugmentedImageDatabase and replace the current active one by calling ArConfig_setAugmentedImageDatabase.

If you want to have 2 databases in your case... you might need to create 2 AugmentedImageDatabases and switch between the two.

**Comment by markus2610** [comment]
@achuvm [Question on Stack Overflow](https://stackoverflow.com/questions/56357696/how-can-i-load-images-later-for-the-ar-augmented-image-database)

**Comment by gom1** [comment]
> @markus2610 yeah, you'll need to recreate the AugmentedImageDatabase and replace the current active one by calling ArConfig_setAugmentedImageDatabase.
> 
> If you want to have 2 databases in your case... you might need to create 2 AugmentedImageDatabases and switch between the two.

I am configuring my session with a new database to which I add the new image I would like to track, and then end up calling ArConfig_setAugmentedImageDatabase before configuring my session. However, I am seeing that the image target can sometimes get in a weird state where both images are tracked even though the database only contains one image.

@achuvm, do you by any chance have any links to documentation specifying how to load new images and swap databases properly?

**Comment by sergeiromanov** [comment]
We need this feature too
